### PR TITLE
Skills

### PR DESCRIPTION
**Pull request by Rohesie**
* Remove the majority of the skill locks, substituting them for ID locks , fumbling time or other penalties.
* Tank access made. Tank is ID locked.
* Dropship had spaghetti locks. Now they are only locked by ID, while lack of pilot skill forbids from optimizing the route and makes operating it slower.
* Riot grenade launchers are ID locked to MPs and Spec weapons to specs. Those with a mind skill can use them without an ID, for Wey-Yu PMCs and other non-marine humans' access is unfinished.
* Smartguns have no locks, but the non-trained user will suffer heavy recoil and accuracy penalties.
* Scouts can use shotguns again.

### CODE REVIEW

**Comment by Rohesie** [comment]
* Everyone can use medevac stretchers and medical HUDs, or see medical reports.
* Everyone can use defibs but at different speeds. Non-trained defibs don't heal (but still revive if the patient is under 200 non-oxy damage), low-skill heals 4 of each damage type per defib, medics heal 8 and doctors 12.
* Everyone can use the autodoc, but they fumble if not trained in surgery.
* Syringes are available to everyone, skill-based speeds unchanged.
* Surgery is noticeably faster for all doctors. Medics take 10 seconds of fumbling, those with basic first-aid and biology knowledge (like SLs, SOs, XOs, COs) take 15, and PFCs 20. Besides being one-second-per-skill-gap slower while actually performing the surgery.

File: code/game/machinery/autodoc.dm [hunk 1]
```diff
@@ -667,9 +671,13 @@
...
-		if(user.mind && user.mind.cm_skills && user.mind.cm_skills.surgery < SKILL_SURGERY_TRAINED && !event)
-			user << "<span class='warning'>You have no idea how to put someone into \the [src]!</span>"
-			return
+		if(user.mind && user.mind.cm_skills && user.mind.cm_skills.surgery < SKILL_SURGERY_PROFESSIONAL && !event)
+			user.visible_message("<span class='notice'>[user] fumbles around figuring out how to put [M] into [src].</span>",
+			"<span class='notice'>You fumble around figuring out how to use put [M] into [src].</span>")
+			var/fumbling_time = SKILL_TASK_TOUGH - ( SKILL_TASK_EASY * ( SKILL_SURGERY_PROFESSIONAL - user.mind.cm_skills.surgery ) ) // 8 secs non-trained, 5 amateur, 2 trained
+			if(!do_after(user, fumbling_time, TRUE, 5, BUSY_ICON_BUILD)) return
+			/*user << "<span class='warning'>You have no idea how to put someone into \the [src]!</span>"
+			return*/
```
Comment (CosmicScientist) [inline code/game/machinery/autodoc.dm]: Remove unused code, don't comment it out. The history of these files are tracked/deleted code can be recovered.

File: code/game/machinery/computer/dropship_weapons.dm [hunk 2]
```diff
@@ -108,9 +108,13 @@
...
-		if(M.mind.assigned_role != "Pilot Officer") //only pilots can fire dropship weapons.
-			usr << "<span class='warning'>A screen with graphics and walls of physics and engineering values open, you immediately force it closed.</span>"
-			return
+		if(M.mind.assigned_role != "Pilot Officer") //everyone can fire dropship weapons while fumbling.
+			usr.visible_message("<span class='notice'>[usr] fumbles around figuring out how to use the automated targeting system.</span>",
+			"<span class='notice'>You fumble around figuring out how to use the automated targeting system.</span>")
+			var/fumbling_time = 100 - 20 * usr.mind.cm_skills.pilot
+			if(!do_after(usr, fumbling_time, TRUE, 5, BUSY_ICON_BUILD)) return
+			/*usr << "<span class='warning'>A screen with graphics and walls of physics and engineering values open, you immediately force it closed.</span>"
+			return*/
```
Comment (CosmicScientist) [inline code/game/machinery/computer/dropship_weapons.dm]: here too

File: code/game/machinery/groundmap_geothermal.dm [hunk 3]
```diff
@@ -104,12 +104,12 @@
 
 	add_fingerprint(user)
 
-	if(user.mind && user.mind.cm_skills && user.mind.cm_skills.engineer < SKILL_ENGINEER_ENGI)
+	/*if(user.mind && user.mind.cm_skills && user.mind.cm_skills.engineer < SKILL_ENGINEER_ENGI)
 		user << "<span class='warning'>You have no clue how this thing works...</span>"
-		return 0
+		return 0*/
```
Comment (CosmicScientist) [inline code/game/machinery/groundmap_geothermal.dm]: beepboop

File: code/game/objects/items/bodybag.dm [hunk 4]
```diff
@@ -292,9 +292,9 @@
 /obj/structure/closet/bodybag/cryobag/Topic(href, href_list)
 	if (href_list["scanreport"])
 		if(hasHUD(usr,"medical"))
-			if(usr.mind && usr.mind.cm_skills && usr.mind.cm_skills.medical < SKILL_MEDICAL_MEDIC)
+			/*if(usr.mind && usr.mind.cm_skills && usr.mind.cm_skills.medical < SKILL_MEDICAL_MEDIC)
 				usr << "<span class='warning'>You're not trained to use this.</span>"
-				return
+				return*/
```
Comment (CosmicScientist) [inline code/game/objects/items/bodybag.dm]: beepboop

File: code/game/objects/items/reagent_containers/autoinjectors.dm [hunk 5]
```diff
@@ -44,11 +45,12 @@
 		reagents.add_reagent("tricordrazine", 15)
 		update_icon()
 
+/*
 /obj/item/reagent_container/hypospray/autoinjector/tricord/skillless
 	name = "\improper First-aid autoinjector"
 	desc = "An autoinjector loaded with a small dose of medicine for marines to treat themselves with."
 	skilllock = 0
-
+*/
```
Comment (CosmicScientist) [inline code/game/objects/items/reagent_containers/autoinjectors.dm]: beepboop

File: code/game/objects/items/reagent_containers/autoinjectors.dm [hunk 6]
```diff
@@ -107,11 +109,12 @@
 		reagents.add_reagent("tramadol", 15)
 		update_icon()
 
+/*
 /obj/item/reagent_container/hypospray/autoinjector/Tramadol/skillless
 	name = "\improper Pain-stop autoinjector"
 	desc = "An auto-injector loaded with a small amount of painkiller for marines to self-administer."
 	skilllock = 0
-
+*/
```
Comment (CosmicScientist) [inline code/game/objects/items/reagent_containers/autoinjectors.dm]: beepboop

File: code/game/objects/items/reagent_containers/syringes.dm [hunk 7]
```diff
@@ -86,11 +86,10 @@
 
 	var/injection_time = 30
 	if(user.mind && user.mind.cm_skills)
-		if(user.mind.cm_skills.medical < SKILL_MEDICAL_MEDIC)
+		/*if(user.mind.cm_skills.medical < SKILL_MEDICAL_MEDIC)
 			user << "<span class='warning'>You aren't trained to use syringes...</span>"
-			return
-		else
-			injection_time = max(5, 50 - 10*user.mind.cm_skills.medical)
+			return*/
```
Comment (CosmicScientist) [inline code/game/objects/items/reagent_containers/syringes.dm]: beepboop

File: code/game/objects/items/stacks/stack.dm [hunk 8]
```diff
@@ -113,10 +113,10 @@
 		var/multiplier = text2num(href_list["multiplier"])
 		if(!multiplier || (multiplier <= 0)) //href exploit protection
 			return
-		if(R.skill_req)
+		/*if(R.skill_req)
 			if(ishuman(usr) && usr.mind && usr.mind.cm_skills && usr.mind.cm_skills.construction < R.skill_req)
 				usr << "<span class='warning'>You are not trained to build this...</span>"
-				return
+				return*/ //Skill check removed
```
Comment (CosmicScientist) [inline code/game/objects/items/stacks/stack.dm]: boopbeep

File: code/game/objects/structures/stool_bed_chair_nest/bed.dm [hunk 9]
```diff
@@ -305,9 +305,9 @@ var/global/list/activated_medevac_stretchers = list()
 	if(!ishuman(user))
 		return
 
-	if(user.mind && user.mind.cm_skills && user.mind.cm_skills.medical < SKILL_MEDICAL_MEDIC)
+	/*if(user.mind && user.mind.cm_skills && user.mind.cm_skills.medical < SKILL_MEDICAL_MEDIC)
 		user << "<span class='warning'>You don't know how to use [src].</span>"
-		return
+		return*/
```
Comment (CosmicScientist) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm]: boopbeep

File: code/modules/clothing/glasses/hud.dm [hunk 10]
```diff
@@ -54,6 +54,7 @@
 	else
 		user << "<span class='warning'>You have no idea what any of the data means and power it off before it makes you nauseated.</span>"
 		active = 0
+*/
```
Comment (CosmicScientist) [inline code/modules/clothing/glasses/hud.dm]: boopbeep

File: code/modules/clothing/suits/marine_armor.dm [hunk 11]
```diff
@@ -316,8 +317,11 @@ var/list/squad_colors = list(rgb(230,25,25), rgb(255,195,45), rgb(200,100,200),
 /obj/item/clothing/suit/storage/marine/specialist/mob_can_equip(mob/M, slot, disable_warning = 0)
 	. = ..()
 	if(.)
-		if(M.mind && M.mind.cm_skills && M.mind.cm_skills.spec_weapons < SKILL_SPEC_TRAINED && M.mind.cm_skills.spec_weapons != SKILL_SPEC_GRENADIER)
+		/*if(M.mind && M.mind.cm_skills && M.mind.cm_skills.spec_weapons < SKILL_SPEC_TRAINED && M.mind.cm_skills.spec_weapons != SKILL_SPEC_GRENADIER)
 			M << "<span class='warning'>You are not trained to use [src]!</span>"
+			return 0*/
```
Comment (CosmicScientist) [inline code/modules/clothing/suits/marine_armor.dm]: boopbeep

File: code/modules/clothing/suits/marine_armor.dm [hunk 12]
```diff
@@ -369,8 +375,11 @@ var/list/squad_colors = list(rgb(230,25,25), rgb(255,195,45), rgb(200,100,200),
 /obj/item/clothing/suit/storage/marine/M3S/mob_can_equip(mob/M, slot, disable_warning = 0)
 	. = ..()
 	if(.)
-		if(M.mind && M.mind.cm_skills && M.mind.cm_skills.spec_weapons < SKILL_SPEC_TRAINED && M.mind.cm_skills.spec_weapons != SKILL_SPEC_SCOUT)
+		/*if(M.mind && M.mind.cm_skills && M.mind.cm_skills.spec_weapons < SKILL_SPEC_TRAINED && M.mind.cm_skills.spec_weapons != SKILL_SPEC_SCOUT)
 			M << "<span class='warning'>You are not trained to use [src]!</span>"
+			return 0*/
```
Comment (CosmicScientist) [inline code/modules/clothing/suits/marine_armor.dm]: beepboop

File: code/modules/clothing/suits/marine_armor.dm [hunk 13]
```diff
@@ -390,8 +400,11 @@ var/list/squad_colors = list(rgb(230,25,25), rgb(255,195,45), rgb(200,100,200),
 /obj/item/clothing/suit/storage/marine/M35/mob_can_equip(mob/M, slot, disable_warning = 0)
 	. = ..()
 	if(.)
-		if(M.mind && M.mind.cm_skills && M.mind.cm_skills.spec_weapons < SKILL_SPEC_TRAINED && M.mind.cm_skills.spec_weapons != SKILL_SPEC_PYRO)
+		/*if(M.mind && M.mind.cm_skills && M.mind.cm_skills.spec_weapons < SKILL_SPEC_TRAINED && M.mind.cm_skills.spec_weapons != SKILL_SPEC_PYRO)
 			M << "<span class='warning'>You are not trained to use [src]!</span>"
+			return 0*/
```
Comment (CosmicScientist) [inline code/modules/clothing/suits/marine_armor.dm]: beepbeep

File: code/modules/cm_marines/dropship_equipment.dm [hunk 14]
```diff
@@ -891,8 +891,12 @@
...
 		return
 	if(user.mind && user.mind.cm_skills && user.mind.cm_skills.pilot < SKILL_PILOT_TRAINED)
-		user << "<span class='warning'> You don't know how to use [src].</span>"
-		return
+		user.visible_message("<span class='notice'>[user] fumbles around figuring out how to use the medevac system.</span>",
+		"<span class='notice'>You fumble around figuring out how to use the medevac system.</span>")
+		var/fumbling_time = 60 - 20 * user.mind.cm_skills.pilot
+		if(!do_after(user, fumbling_time, TRUE, 5, BUSY_ICON_BUILD)) return
+		/*user << "<span class='warning'> You don't know how to use [src].</span>"
+		return*/
```
Comment (CosmicScientist) [inline code/modules/cm_marines/dropship_equipment.dm]: boopboop

File: code/modules/mob/living/carbon/human/examine.dm [hunk 15]
```diff
@@ -569,8 +569,8 @@
 				if(H.mind && (!H.mind.cm_skills || H.mind.cm_skills.police >= SKILL_POLICE_MP))
 					return istype(H.glasses, /obj/item/clothing/glasses/hud/security) || istype(H.glasses, /obj/item/clothing/glasses/sunglasses/sechud)
 			if("medical")
-				if(H.mind && H.mind.cm_skills && H.mind.cm_skills.medical >= SKILL_MEDICAL_MEDIC)
-					return istype(H.glasses, /obj/item/clothing/glasses/hud/health)
+				/*if(H.mind && H.mind.cm_skills && H.mind.cm_skills.medical >= SKILL_MEDICAL_MEDIC)
+					*/return istype(H.glasses, /obj/item/clothing/glasses/hud/health)
```
Comment (CosmicScientist) [inline code/modules/mob/living/carbon/human/examine.dm]: beeeeep

File: code/modules/mob/living/carbon/human/human.dm [hunk 16]
```diff
@@ -805,9 +805,9 @@
 									R.fields[text("com_[counter]")] = text("Made by [U.name] ([U.modtype] [U.braintype]) on [time2text(world.realtime, "DDD MMM DD hh:mm:ss")], [game_year]<BR>[t1]")
 
 	if (href_list["medholocard"])
-		if(usr.mind && usr.mind.cm_skills && usr.mind.cm_skills.medical < SKILL_MEDICAL_MEDIC)
+		/*if(usr.mind && usr.mind.cm_skills && usr.mind.cm_skills.medical < SKILL_MEDICAL_MEDIC)
 			usr << "<span class='warning'>You're not trained to use this.</span>"
-			return
+			return*/
```
Comment (CosmicScientist) [inline code/modules/mob/living/carbon/human/human.dm]: beepboop

File: code/modules/mob/living/carbon/human/human.dm [hunk 17]
```diff
@@ -827,9 +827,9 @@
 
 	if (href_list["scanreport"])
 		if(hasHUD(usr,"medical"))
-			if(usr.mind && usr.mind.cm_skills && usr.mind.cm_skills.medical < SKILL_MEDICAL_MEDIC)
+			/*if(usr.mind && usr.mind.cm_skills && usr.mind.cm_skills.medical < SKILL_MEDICAL_MEDIC)
 				usr << "<span class='warning'>You're not trained to use this.</span>"
-				return
+				return*/
```
Comment (CosmicScientist) [inline code/modules/mob/living/carbon/human/human.dm]: beepboop

File: code/modules/projectiles/updated_projectiles/guns/flamer.dm [hunk 18]
```diff
@@ -340,9 +342,9 @@
 		if(!current_mag || !current_mag.current_rounds)
 			return
 
-		if(user.mind && user.mind.cm_skills && user.mind.cm_skills.spec_weapons < SKILL_SPEC_TRAINED && user.mind.cm_skills.spec_weapons != SKILL_SPEC_PYRO)
+		/*if(user.mind && user.mind.cm_skills && user.mind.cm_skills.spec_weapons < SKILL_SPEC_TRAINED && user.mind.cm_skills.spec_weapons != SKILL_SPEC_PYRO)
 			user << "<span class='warning'>You don't seem to know how to use [src]...</span>"
-			return 0
+			return 0*/ //GUN_SPECIALIST deals with this
```
Comment (CosmicScientist) [inline code/modules/projectiles/updated_projectiles/guns/flamer.dm]: beepboop

File: code/modules/projectiles/updated_projectiles/guns/shotguns.dm [hunk 19]
```diff
@@ -145,12 +145,12 @@ can cause issues with ammo types getting mixed up during the burst.
...
 
 
-/obj/item/weapon/gun/shotgun/able_to_fire(mob/living/user)
+/*/obj/item/weapon/gun/shotgun/able_to_fire(mob/living/user)
 	. = ..()
 	if (. && istype(user)) //Let's check all that other stuff first.
 		if(user.mind && user.mind.cm_skills && user.mind.cm_skills.spec_weapons == SKILL_SPEC_SCOUT)
 			user << "<span class='warning'>Scout specialists can't use shotguns...</span>"
-			return 0
+			return 0*/ //Let's remove this artificial restriction.
```
Comment (CosmicScientist) [inline code/modules/projectiles/updated_projectiles/guns/shotguns.dm]: beepboop

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 20]
```diff
@@ -9,12 +9,12 @@
...
 	wield_delay = WIELD_DELAY_SLOW
 
-	able_to_fire(mob/living/user)
+	/*able_to_fire(mob/living/user)
 		. = ..()
 		if(. && istype(user)) //Let's check all that other stuff first.
 			if(user.mind && user.mind.cm_skills && user.mind.cm_skills.spec_weapons < SKILL_SPEC_TRAINED && user.mind.cm_skills.spec_weapons != SKILL_SPEC_SNIPER)
 				user << "<span class='warning'>You don't seem to know how to use [src]...</span>"
-				return 0
+				return 0*/ // GUN_SPECIALIST does the job
```
Comment (CosmicScientist) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]: beepboop

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 21]
```diff
@@ -190,12 +192,12 @@
...
 	recoil = config.min_recoil_value
 
-/obj/item/weapon/gun/rifle/m4ra/able_to_fire(mob/living/user)
+/*/obj/item/weapon/gun/rifle/m4ra/able_to_fire(mob/living/user)
 	. = ..()
 	if (. && istype(user)) //Let's check all that other stuff first.
 		if(user.mind && user.mind.cm_skills && user.mind.cm_skills.spec_weapons < SKILL_SPEC_TRAINED && user.mind.cm_skills.spec_weapons != SKILL_SPEC_SCOUT)
 			user << "<span class='warning'>You don't seem to know how to use [src]...</span>"
-			return 0
+			return 0*/ // GUN_SPECIALIST does the job
```
Comment (CosmicScientist) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]: beepboop

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 22]
```diff
@@ -255,9 +257,9 @@
 	if(.)
 		if(!ishuman(user)) return 0
 		var/mob/living/carbon/human/H = user
-		if(user.mind && user.mind.cm_skills && user.mind.cm_skills.smartgun < SKILL_SMART_USE)
+		/*if(user.mind && user.mind.cm_skills && user.mind.cm_skills.smartgun < SKILL_SMART_USE)
 			user << "<span class='warning'>You don't seem to know how to use [src]...</span>"
-			return 0
+			return 0 */ // Everyone can use the smartgun.
```
Comment (CosmicScientist) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]: beepboop

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 23]
```diff
@@ -405,12 +409,12 @@
...
 	else user << "<span class='warning'>It's empty!</span>"
 
-/obj/item/weapon/gun/launcher/m92/able_to_fire(mob/living/user)
+/*/obj/item/weapon/gun/launcher/m92/able_to_fire(mob/living/user)
 	. = ..()
 	if (. && istype(user)) //Let's check all that other stuff first.
 		if(user.mind && user.mind.cm_skills && user.mind.cm_skills.spec_weapons < SKILL_SPEC_TRAINED && user.mind.cm_skills.spec_weapons != SKILL_SPEC_GRENADIER)
 			user << "<span class='warning'>You don't seem to know how to use [src]...</span>"
-			return 0
+			return 0*/ // GUN_SPECIALIST does the job
```
Comment (CosmicScientist) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]: beepboop

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 24]
```diff
@@ -530,7 +535,7 @@
 					return 0
 			else if(user.mind.cm_skills.spec_weapons < SKILL_SPEC_TRAINED && user.mind.cm_skills.spec_weapons != SKILL_SPEC_GRENADIER)
 				user << "<span class='warning'>You don't seem to know how to use [src]...</span>"
-				return 0
+				return 0*/ // GUN_SPECIALIST and GUN_POLICE do the job
```
Comment (CosmicScientist) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]: beepboop

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 25]
```diff
@@ -609,15 +617,15 @@
...
+	/*if (. && istype(user)) //Let's check all that other stuff first.
 		/*var/turf/current_turf = get_turf(user)
 		if (current_turf.z == 3 || current_turf.z == 4) //Can't fire on the Almayer, bub.
 			click_empty(user)
 			user << "<span class='warning'>You can't fire that here!</span>"
 			return 0*/
 		if(user.mind && user.mind.cm_skills && user.mind.cm_skills.spec_weapons < SKILL_SPEC_TRAINED && user.mind.cm_skills.spec_weapons != SKILL_SPEC_ROCKET)
 			user << "<span class='warning'>You don't seem to know how to use [src]...</span>"
-			return 0
+			return 0*/ //GUN_SPECIALIST does the job
```
Comment (CosmicScientist) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]: beepboop

File: code/modules/shuttles/shuttle_console.dm [hunk 26]
```diff
@@ -198,16 +198,19 @@
...
-				return
-			else
-				usr << "<span class='notice'>You upload a flight plan for a low altitude flyby above the planet.</span>"
+			if(usr.mind && usr.mind.cm_skills && usr.mind.cm_skills.pilot < SKILL_PILOT_TRAINED) //everyone can activate the fire mission mode while fumbling, but everyone can reset it back to transport without.
+				usr.visible_message("<span class='notice'>[usr] fumbles around figuring out how to set the autopilot.</span>",
+				"<span class='notice'>You fumble around figuring out how to set the autopilot.</span>")
+				var/fumbling_time = 100 - 20 * usr.mind.cm_skills.pilot
+				if(!do_after(usr, fumbling_time, TRUE, 5, BUSY_ICON_BUILD)) return
+				/*usr << "<span class='warning'>A screen with graphics and walls of physics and engineering values open, you immediately force it closed.</span>"
+				return*/
```
Comment (CosmicScientist) [inline code/modules/shuttles/shuttle_console.dm]: beepboop

File: code/modules/surgery/surgery.dm [hunk 27]
```diff
@@ -100,9 +100,13 @@ proc/do_surgery(mob/living/carbon/M, mob/living/user, obj/item/tool)
...
-	if(user.mind && user.mind.cm_skills && user.mind.cm_skills.surgery < SKILL_SURGERY_TRAINED)
-		user << "<span class='warning'>You have no idea how to do surgery...</span>"
-		return 1
+	if(user.mind && user.mind.cm_skills && user.mind.cm_skills.surgery < SKILL_SURGERY_PROFESSIONAL)
+		user.visible_message("<span class='notice'>[user] fumbles around figuring out how to operate [M].</span>",
+		"<span class='notice'>You fumble around figuring out how to operate [M].</span>")
+		var/fumbling_time = SKILL_TASK_FORMIDABLE - ( SKILL_TASK_AVERAGE * user.mind.cm_skills.surgery ) // 20 secs non-trained, 15 amateur, 10 semi-prof
+		if(!do_after(user, fumbling_time, TRUE, 5, BUSY_ICON_BUILD)) return
+		/*user << "<span class='warning'>You have no idea how to do surgery...</span>"
+		return 1*/
```
Comment (CosmicScientist) [inline code/modules/surgery/surgery.dm]: beepboop

File: code/modules/vehicles/powerloader.dm [hunk 28]
```diff
@@ -78,9 +78,9 @@
 	if(M != user) return
 	if(!ishuman(M))	return
 	var/mob/living/carbon/human/H = M
-	if(H.mind && user.mind.cm_skills && !user.mind.cm_skills.powerloader)
+	/*if(H.mind && user.mind.cm_skills && !user.mind.cm_skills.powerloader)
 		H << "<span class='warning'>You don't seem to know how to operate [src].</span>"
-		return
+		return*/
```
Comment (CosmicScientist) [inline code/modules/vehicles/powerloader.dm]: boopbeep

File: code/modules/vehicles/tank.dm [hunk 29]
```diff
@@ -209,10 +211,18 @@
...
+		M << "<span class='warning'>Access denied.</span>"
 		return
 
+	if(!M.mind || !(!M.mind.cm_skills || M.mind.cm_skills.large_vehicle >= SKILL_LARGE_VEHICLE_TRAINED))
+		M.visible_message("<span class='notice'>[M] fumbles around figuring out how to get into [src].</span>",
+		"<span class='notice'>You fumble around figuring out how to get into [src].</span>")
+		var/fumbling_time = 100 - 20 * M.mind.cm_skills.large_vehicle
+		if(!do_after(M, fumbling_time, TRUE, 5, BUSY_ICON_BUILD)) return
+		/*M << "<span class='notice'>You have no idea how to operate this thing.</span>"
+		return*/
```
Comment (CosmicScientist) [inline code/modules/vehicles/tank.dm]: and that should be all of them.

**Comment by ninjanomnom** [comment]
What cosmic said, also this is the sort of thing I'd want to testmerge before we fully merge. Seems mostly fine though.

**Comment by Iatots** [comment]
How do you fumble on the autodoc console? Do you add unnecessary surgeries to the queue? Remove previously entered surgeries?

Speaking of, maybe it's OOS but could you add removing queued up surgeries from autodoc?

**Comment by Rohesie** [comment]
You simply take long to put someone into the autodoc. Could be added that you also fumble while using it, but it wouldn't make sense for the machine to work slower because you are not skilled. Your use of it is the part that is impaired, only.

You can remove queued up surgeries. What do you mean?

**Comment by Rohesie** [comment]
Any further issues with this?

**Comment by Iatots** [comment]
Please give tanker boys lv2 engineering so they can take care of the best themselves.

File: code/__DEFINES/access.dm [hunk 30]
```diff
@@ -31,6 +31,7 @@ most of them are tied into map-placed objects. This should be reworked in the fu
 #define ACCESS_MARINE_PILOT 		23
 #define ACCESS_MARINE_WO			24
 #define ACCESS_MARINE_RO			26
+#define ACCESS_MARINE_TANK			30
```
Comment (ninjanomnom) [inline code/__DEFINES/access.dm]: 27

**Pull request merged**